File: API/src/dtos/registerFacilityDto.ts
import { FileDto } from "./fileDto";

export interface RegisterFacilityDto {
  userId?: string;
  facilityName: string;
  lat: number;
  lon: number;
  description: string;
  policy: string;
  detailAddress: string;
  province: string;
  fullName: string;
  email: string;
  phoneNumber: string;
  citizenId: string;
  facebookUrl?: string | undefined;
  facilityImages?: FileDto[];
  citizenImageFront?: FileDto;
  citizenImageBack?: FileDto;
  bankCardFront?: FileDto;
  bankCardBack?: FileDto;
  businessLicenseImages?: FileDto[];
}

File: BadcourtAPI/FacilityService/FacilityService.Core/FacilityService.Core.Application/DTOs/RegisterFacilityDto.cs
using Microsoft.AspNetCore.Http;

namespace FacilityService.Core.Application.DTOs;

public class RegisterFacilityDto
{
    public required string FacilityName { get; set; }
    public double Lat { get; set; }
    public double Lon { get; set; }
    public required string Description { get; set; }
    public required string Policy { get; set; }
    public required string DetailAddress { get; set; }
    public required string Province { get; set; }
    public required string FullName { get; set; }
    public required string Email { get; set; }
    public required string PhoneNumber { get; set; }
    public required string CitizenId { get; set; }
    public string? FacebookUrl { get; set; }
    public List<IFormFile> FacilityImages { get; set; } = [];
    public required IFormFile CitizenImageFront { get; set; }
    public required IFormFile CitizenImageBack { get; set; }
    public required IFormFile BankCardFront { get; set; }
    public required IFormFile BankCardBack { get; set; }
    public List<IFormFile> BusinessLicenseImages { get; set; } = [];
}
